### DISCUSSION TITLE
Custom types and convertToDatabaseValue behavior

### DISCUSSION THREAD

**Discussion by ccpu** [mikro-orm/mikro-orm#6206] [General]
Hello,

I am currently working with MikroORM and have implemented a custom type called `UuidBinaryType` to handle UUIDs stored as binary in my database. I encountered an issue where the `convertToDatabaseValue` method is called twice during a `nativeUpdate` operation; once with the UUID string and once with the already converted Buffer.

Here’s the relevant part of my code:

```ts
@Entity({ tableName: 'project_files' })
export class File {
  @PrimaryKey({
    type: UuidBinaryType,
  })
  id!: string; // Store as Buffer
}
```
And here’s the custom type implementation:

```ts
export class UuidBinaryType extends Type<string, Buffer> {
  convertToDatabaseValue(value: string): Buffer {
    return Buffer.from(uuidParse(value) as any);
  }

  convertToJSValue(value: Buffer): string {
    return uuidStringify(value as any);
  }

  getColumnType(): string {
    return 'binary(16)';
  }
}
```
The first call to `convertToDatabaseValue` receives the UUID string, which is expected. However, the second call receives a Buffer, which leads to an error if not handled properly.

I modified the method to check the type of the value, which resolved the issue:

```ts
convertToDatabaseValue(value: string): Buffer {
  if (typeof value !== 'string') return value; // Prevent double conversion
  return Buffer.from(uuidParse(value) as any);
}
```
Wondring why is `convertToDatabaseValue` called twice during the `nativeUpdate` operation? Is this behavior expected, or could it indicate a potential issue in how I am using MikroORM?

Any insights or guidance would be greatly appreciated!

**Comment by B4nan** [comment]
Sounds like a bug, hard to guess, since, you haven't shared code for the most important part - what you are doing. 

[Complete reproduction](https://github.com/mikro-orm/reproduction) welcome.

**Comment by ccpu** [comment]
I have created a reproduction as you requested. In this reproduction repo, the test will throw an error due to the issue mentioned above.

[reproduction-custom-type-issue](https://github.com/ccpu/reproduction-custom-type-issue)

Thanks

**Reply by B4nan** [reply]
Yeah, that is surely a bug, this should help: 1b398465d4c9b1ef76983ebefd50cda0d344ff8e

**Reply by ccpu** [reply]
Thank you for the quick fix!
